### PR TITLE
Fix matrix-project hard dependency

### PR DESCRIPTION
**Pull request by timja**
Missed a call to matrix-project compile time dependency before

Closes https://github.com/jenkinsci/slack-plugin/issues/557

### CODE REVIEW

File: src/main/java/jenkins/plugins/slack/ActiveNotifier.java [hunk 1]
```diff
@@ -154,7 +153,7 @@ public void completed(AbstractBuild r) {
     }
 
     private boolean skipOnMatrixChildren(AbstractBuild build) {
-        return build instanceof MatrixRun && !notifier.getMatrixTriggerMode().forChild;
+        return notifier.isMatrixProject(build) && !notifier.getMatrixTriggerMode().forChild;
```
Comment (casz) [inline src/main/java/jenkins/plugins/slack/ActiveNotifier.java]: You're missing the other problem, getMatrixTriggerMode can return null :laughing:
`private MatrixTriggerMode matrixTriggerMode = MatrixTriggerMode.ONLY_CONFIGURATIONS;`
The should be no need for having it set on non-Matrix jobs either.

Comment (casz) [reply src/main/java/jenkins/plugins/slack/ActiveNotifier.java]: Give me commit access than I have a fix for that :sweat_smile:

Comment (timja) [reply src/main/java/jenkins/plugins/slack/ActiveNotifier.java]: done

Comment (casz) [reply src/main/java/jenkins/plugins/slack/ActiveNotifier.java]: That should do it :sweat:

Comment (casz) [reply src/main/java/jenkins/plugins/slack/ActiveNotifier.java]: I swear the code working before I merged it with yours :rofl:

Comment (timja) [reply src/main/java/jenkins/plugins/slack/ActiveNotifier.java]:
```
[2019-04-15T13:22:06.034Z] [ERROR] Possible null pointer dereference in jenkins.plugins.slack.ActiveNotifier.skipOnMatrixChildren(AbstractBuild) due to return value of called method [jenkins.plugins.slack.ActiveNotifier, jenkins.plugins.slack.ActiveNotifier] Dereferenced at ActiveNotifier.java:[line 157]Known null at ActiveNotifier.java:[line 157] NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE

```

Comment (casz) [reply src/main/java/jenkins/plugins/slack/ActiveNotifier.java]: I hate mocking! The isMatrixRun is being mocked out...

Comment (casz) [reply src/main/java/jenkins/plugins/slack/ActiveNotifier.java]: SpotBugs on my end does not complain :confused:

Comment (timja) [reply src/main/java/jenkins/plugins/slack/ActiveNotifier.java]: fails for me with 
```
mvn spotbugs:checl
```

**Comment by casz** [comment]
@timja GitHub being odd: https://github.com/timja/slack-plugin/commit/cdfc98ece3e656e7634825171d79c90b110760a6 yet PR is not updated :sob:

**Comment by casz** [comment]
Jenkins CI being slow :cry:

**Pull request merged**